main.tf

```diff
@@ -0,0 +1,39 @@
+/**
+ * # AWS EKS Load Balancer Controller Terraform module
+ *
+ * A Terraform module to deploy the AWS Load Balancer Controller on Amazon EKS cluster.
+ *
+ * [![Terraform validate](https://github.com/lablabs/terraform-aws-eks-load-balancer-controller/actions/workflows/validate.yaml)
+ * [![pre-commit](https://github.com/lablabs/terraform-aws-eks-load-balancer-controller/actions/workflows/pre-commit.yml)
+ */
+locals {
+  addon = {
+    name = "aws-load-balancer-controller"
+
+    helm_chart_version = "1.12.0"
+    helm_repo_url      = "https://aws.github.io/eks-charts"
+  }
+
+  addon_irsa = {
+    (local.addon.name) = {
+      irsa_policy_enabled         = local.irsa_policy_enabled
+      irsa_policy                 = (var.enabled && var.irsa_policy != null) ? var.irsa_policy : data.aws_iam_policy_document.this[0].json
```

Comment:
Please, check this with Ati so we are aligned how we make the condition here

Reply:
@atyu1 condition modified to be the same as in terraform-aws-eks-external-dns addon. Is it OK?

Reply:
It should be enough according to https://github.com/lablabs/terraform-aws-eks-cluster-autoscaler/pull/23#discussion_r2092377952